ipinfo/cmd_init.go

```diff
@@ -0,0 +1,301 @@
+	},
+}
+
+func printHelpInit() {
+	fmt.Printf(
+		`Usage: %s init [<opts>] [<token>]
+Examples:
+	# Login command with token flag.
+	$ init --token <token>
+
+	#Authentication without token flag.
+	$ init
+
+Options:
+  --token <tok>, -t <tok>
+    token to login with.
+    (this is potentially unsafe; let the CLI prompt you instead).
+  --no-check
+    disable checking if the token is valid or not.
+    default: false.
+  --help, -h
+    show help.
+`, progBase)
+}
+
+type signupCli struct {
+	SignupURL string `json:"signupURL"`
+}
+
+type tokenCli struct {
+	Token string `json:"token"`
+}
+
+type meResponse struct {
+	Error string `json:"error"`
+}
+
+func cmdInit() error {
+	var opt int
+	var fTok string
+	var fNoCheck bool
+
+	pflag.StringVarP(&fTok, "token", "t", "", "the token to save.")
+	pflag.BoolVar(&fNoCheck, "no-check", false, "disable checking if token is valid.")
+	pflag.BoolVarP(&fHelp, "help", "h", false, "show help.")
+	pflag.Parse()
+
+	if fHelp {
+		printHelpInit()
+		return nil
+	}
+
+	// get args without subcommand.
+	args := pflag.Args()[1:]
+
+	// only token arg allowed.
+	if len(args) > 1 {
+		printHelpInit()
+		return nil
+	}
+
+	// allow only flag or arg for token but not both.
+	if fTok != "" && len(args) > 0 {
+		return errors.New("ambiguous token input source")
+	}
+
+	// get token, from flag or command line.
+	// if it exists, we'll exit early as it's an implicit login.
+	tok := fTok
+	if len(args) > 0 {
+		tok = args[0]
+	}
+	if tok != "" {
+		if !fNoCheck {
+			if err := checkValidity(tok); err != nil {
+				return fmt.Errorf("could not confirm if token is valid: %w", err)
+			}
+		}
+
+		// save token to file.
+		gConfig.Token = tok
+		if err := SaveConfig(gConfig); err != nil {
+			return err
+		}
+
+		fmt.Println("done")
+
+		return nil
+	}
+
+	fmt.Println("1) Enter an existing API token")
+	fmt.Println("2) Create a new account")
+	_, err := fmt.Scanf("%d", &opt)
+	if err != nil {
+		return fmt.Errorf("error reading input: %v", err)
+	}
+	if opt == 1 {
+		newtoken, err := enterToken(tok)
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+
+		// check token validity.
+		if !fNoCheck {
+			if err := checkValidity(newtoken); err != nil {
+				return fmt.Errorf("could not confirm if token is valid: %w", err)
+			}
+		}
+
+		// save token to file.
+		gConfig.Token = tok
+		if err := SaveConfig(gConfig); err != nil {
+			return err
+		}
+
+		fmt.Println("done")
+
+		return nil
+	} else if opt == 2 {
+		res, err := http.Get("https://ipinfo.io/signup/cli")
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
+
+		// parse response.
+		rawBody, err := io.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+		body := &signupCli{}
+		err = json.Unmarshal(rawBody, body)
+		if err != nil {
+			return err
+		}
+
+		err = openURL(body.SignupURL)
+		fmt.Println("If the link does not open, please go to this link to get your access token:")
+		fmt.Printf("%v\n", body.SignupURL)
```

Comment:
Add one extra newline:

```suggestion
		fmt.Println("")
		fmt.Printf("%v\n", body.SignupURL)
```